packages/analytics/index.tsx

```diff
@@ -1,54 +1,2 @@
-import { AddToCartInput, CartLineItem } from '@ifixit/cart-sdk';
-import { gaSendEvent, trackGoogleAddToCart } from './google';
-import { trackPiwikCartChange } from './piwik';
-import { TrackEventPiwik, trackInPiwik } from './piwik/track-event';
-
 export * from './google';
 export * from './piwik';
-
-/**
- * @param trackData trackData.eventName will default to the page path if not provided
- */
-export const trackInPiwikAndGA = (trackData: TrackEventPiwik) => {
-   if (typeof window === 'undefined') {
-      return;
-   }
-   const eventName =
-      trackData.eventName ||
-      `${window.location.origin}${window.location.pathname}`;
```

Comment:
This is the only thing I'm sad to see go.

Previously, this was optional and set to the default Max asked for. Now you must repeat it at each callsite

Reply:
Yeah its a little messier but i felt it was the best decision. Do you think theres is a cleaner solution?

Reply:
Yea, I think a small wrapper around pushToPiwik would be nice:

```
export type TrackEvent = {
   /**
    * Describes the type of events you want to track.
    * For example, Link Clicks, Videos, Outbound Links, and Form Events.
    */
   eventCategory: string;
   /**
    * The specific action that is taken.
    * For example, with a Video category, you might have a Play, Pause and Complete action.
    */
   eventAction: string;
   /**
    * Usually the title of the element that is being interacted with, to aid with analysis.
    * For example, it could be the name of a Video that was played or the specific
    * form that is being submitted.
    */
   eventName?: string;
   /**
    * A numeric value and is often added dynamically. It could be the cost of a
    * product that is added to a cart, or the completion percentage of a video.
    */
   eventValue?: number;
};


export const trackEventInPiwik = (trackData: TrackEvent) => {
   pushToPiwik([
      'trackEvent',
      trackData.eventCategory,
      trackData.eventAction,
      trackData.eventName,
      trackData.eventValue,
   ]);
};
```

The main benefit is that we get some extra type info.

I don't know if this holds true for Piwik, but Matomo actually responded with a 200 (data sent) but threw out any data that did not match the types.

Example:

```
pushToPiwik([
      'trackEvent',
      '',
      '',
      '',
      "1234",
   ]);
```

The fact we send a string caused Matomo to toss the whole event and not be recorded.

Reply:
i might do this in my consolidation pull